File: PT3/PE_SBA301_SP25_FE_TranDinhDuyPhuong/src/services/carService.js
import api from '../api/axiosConfig';

const carService = {
    getAllCars: async () => {
        return await api.get('/cars');
    },

    addCar: async (carData) => {
        return await api.post('/cars', carData);
    },

    deleteCar: async (carId) => {
        return await api.delete(`/cars/${carId}`);
    }
};

export default carService;

File: PT3/PE_SBA301_SP25_BE_DE180212/src/main/java/sba301/pe_sba301_sp25_be_de180212/services/CarService.java
package sba301.pe_sba301_sp25_be_de180212.services;

import sba301.pe_sba301_sp25_be_de180212.dtos.cars.CarRequest;
import sba301.pe_sba301_sp25_be_de180212.dtos.cars.CarResponse;

import java.util.List;

public interface CarService {

    CarResponse addCar(CarRequest carRequest);

    CarResponse updateCar(Integer carId, CarRequest carRequest);

    CarResponse deleteCar(Integer carId);

    List<CarResponse> getAllCars();
}
